src/main/java/uk/gov/register/filters/HttpToHttpsRedirectFilter.java

```diff
@@ -0,0 +1,44 @@
+package uk.gov.register.filters;
+
+import com.google.common.net.HttpHeaders;
+
+import javax.servlet.*;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.ext.Provider;
+import java.io.IOException;
+
+@Provider
+public class HttpToHttpsRedirectFilter implements javax.servlet.Filter {
+
+   @Override
+   public void init(FilterConfig filterConfig) throws ServletException { }
+
+   @Override
+   public void destroy() { }
+
+   @Override
+   public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
+      HttpServletRequest httpRequest = (HttpServletRequest)request;
+
+      if ("http".equals(httpRequest.getHeader(HttpHeaders.X_FORWARDED_PROTO))) {
+         StringBuilder location = new StringBuilder();
```

Comment:
httpRequest.getRequestURL() returns the complete url except the query string as a StringBuffer - but that would only save a couple of lines.